crypto/encrypting_reader_test.go

```diff
@@ -0,0 +1,37 @@
+package crypto
+
+import (
+	"bytes"
+	"testing"
+	"testing/quick"
+)
+
+func encryptingReaderCanReadDifferentSizeOfChunks(in []byte, bufferSize uint8) bool {
+	inPadded := padWithPKCS7VarBlock(in, 16)
+	if bufferSize == 0 {
+		return true
+	}
+	encryptingReader := newBlockModeEncryptingReader(bytes.NewReader(in), &DoNothingBlockMode{})
+	readDataBuffer := bytes.NewBuffer(nil)
+	buffer := make([]byte, bufferSize)
+	numberOfReadBytes := 0
+
+	var e error
+	var readBytes int
+
+	for e == nil {
```

Comment:
I'm assuming that it waits until `encryptingReader.Read()` will return `error.EOF`, am I right?

Maybe it would be better to wait for concrete error?
What will happen if any other error will appear? Will we return `false` in case of any error?

Reply:
I will end the loop for any error just like that but return false for any error other than EOF. Is that alright?